### ISSUE TITLE
Can't get Sensordata to run

### ISSUE THREAD

**Issue by flo269** [alexstocker/sensordata#1]
Hi,

I just I can't get the script below executed. I get the following Error:
> $ python sensordata.py 22 4
<img width="570" alt="screen shot 2017-07-06 at 12 35 03" src="https://user-images.githubusercontent.com/25147331/27907354-9b199744-6247-11e7-8503-80cf483c0f2a.png">

I have checked and there are no spaces. All are tabs...

Any help is highly appreciated!
Thanks :-)

```python
#!/usr/bin/python
import sys
import time
import Adafruit_DHT
import threading
import json
import urllib2
import base64
import pprint
from datetime import datetime

start_time = time.time()
interval = 10 # set interval in seconds
deviceId = 'REMOVED' # for example: a uuid like this f7645058-fe8c-11e6-bc64-92361f002671
url = 'https://REMOVED.IT/nextcloud/index.php/apps/sensorlogger/api/v1/createlog/' # for example: https://YOUR-OWNCLOUD-INSTANCE/index.php/apps/sensorlogge$
headers = {'content-type': 'application/json'}

sensor_args = { '11': Adafruit_DHT.DHT11,
				'22': Adafruit_DHT.DHT22,
				'2302': Adafruit_DHT.AM2302 }
if len(sys.argv) == 3 and sys.argv[1] in sensor_args:
	sensor = sensor_args[sys.argv[1]]
	pin = sys.argv[2]
else:
	print('usage: sudo ./Adafruit_DHT.py [11|22|2302] GPIOpin#')
	print('example: sudo ./Adafruit_DHT.py 2302 4 - Read from an AM2302 connected to GPIO #4')
	sys.exit(1)

def sensorData():
	threading.Timer(interval, sensorData).start()
	humidity, temperature = Adafruit_DHT.read_retry(sensor, pin)
	cdatetime = datetime.now()
	currentDate = cdatetime.strftime('%Y-%m-%d %H:%M:%S')

	if humidity is not None and temperature is not None:
				payload = {
						'deviceId': deviceId,
						'temperature': temperature,
						'humidity': humidity,
						'date': currentDate
						}
		req = urllib2.Request(url)
				base64string = base64.encodestring('%s:%s' % ('SOMEUSERNAE', 'SOMEPASSWORD')).replace('\n', '')
				req.add_header("Authorization", "Basic %s" % base64string)
				req.add_header("Content-Security-Policy", "default-src 'none';script-src 'self' 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self$
		req.add_header('Content-Type','application/json')
		data = json.dumps(payload)
		data = json.dumps(payload)
				response = urllib2.urlopen(req,data)
		else:
		print('Failed to get reading. Try again!')
		sys.exit(1)

sensorData()
```

**Comment by flo269** [comment]
Python 2.7.9 and Raspbian GNU/Linux 8 (jessie)

**Comment by flo269** [comment]
I copied the script from the repo again and just added my url and device id. I used sublime to (view-> indentation -> convert indentation to tabs) convert everything to tabs and now the error is gone.

BUT now I get the following :-) :

$ python sensordata.py 22 4
Traceback (most recent call last):
  File "sensordata.py", line 55, in <module>
    sensorData()
  File "sensordata.py", line 50, in sensorData
    response = urllib2.urlopen(req,data)
  File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 401: Unauthorized

The device is registered and listed in the app on nextcloud. Any Ideas?
